R/utils_report.R

```diff
@@ -1,73 +1,118 @@
-#' Create a performance report for the tests that were run
+#' @title Combine list of performances in a single data.frame
+#'
+#' @param performance_list list of tests containing commits with dates, duration
+#' time and test name
+#'
+#' @export
+combine_performances <- function(performance_list) {
+  # create an unique data.frame for all branches and repetitions
+  df_all <- mapply(
+    performance_list,
+    names(performance_list),
+    FUN = function(x, y) {
+      df <- bind_rows(x)
+      df$branch <- y
+
+      return(df)
+    },
+    SIMPLIFY = FALSE
+  )
+
+  # bind rows
+  df_all <- bind_rows(df_all)
+
+  # return a single data.frame
+  return(df_all)
+}
+
+#' @title Create a performance report for the tests that were run
 #'
 #' @param report_params list of tests containing commits with dates, duration
 #' time and test name
-#' @param report_name name of the file to which the report should be saved, without
-#' the extension
-#' @param report_dir name of the folder where the report should be saved
+#' @param file name of the file to which the report should be saved (.html)
 #'
 #' @importFrom quarto quarto_render
-#' @importFrom rstudioapi selectDirectory
 #' @export
```

Comment:
A function's documentation must always include `@examples` and `@return`, even if it is called for side effects (CRAN is very strict about this). Check [this](https://github.com/Appsilon/rhino/blob/main/R/tools.R#L5) example from Rhino. Please apply to all functions.

Reply:
Thanks for the comment, @jakubnowicki. I added @examples and @return for all functions in report_utils.R and created a new issue to add the documentation to other files in the project: https://github.com/Appsilon/shiny.benchmark/issues/111